frontend/src/reusable/inputs/ListBoxPrompt.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from "react";
+import { SimplePromptDefinition } from "../PromptTypes";
+import {
+    Listbox,
+    ListboxButton,
+    ListboxOption,
+    ListboxOptions,
+    Transition,
+} from "@headlessui/react";
+import { ChevronDownIcon, CheckIcon } from "@heroicons/react/20/solid";
+
+export default function ListBoxPrompt(props: SimplePromptDefinition) {
+    let currentSelected = props.state.selected;
+    return (
+        <Listbox
+            onChange={(e) => {
+                props.stateManager(e);
+                if (props.doSomething) props.doSomething(e);
+            }}
```

Comment:
yucky!

Reply:
if you make doSomething non optional, you don't have to do this anymore

Reply:
honestly i think passing down this function is wrong in the first place and theres probably a better way of doing it but youre right ill make it required